libflux/flux-core/src/semantic/tests/labels.rs

```diff
@@ -0,0 +1,226 @@
+use super::*;
+
+use crate::semantic::Feature;
+
+#[test]
+fn labels_simple() {
+    test_infer! {
+        config: AnalyzerConfig{
+            features: vec![Feature::LabelPolymorphism],
+            ..AnalyzerConfig::default()
+        },
+        env: map![
+            "fill" => "(<-tables: [{ A with B: C }], ?column: B, ?value: D) => [{ A with B: D }]
+                where B: Label
+                "
+        ],
+        src: r#"
+            x = [{ a: 1 }] |> fill(column: "a", value: "x")
+            y = [{ a: 1, b: ""}] |> fill(column: "b", value: 1.0)
+            b = "b"
+            z = [{ a: 1, b: ""}] |> fill(column: b, value: 1.0)
+        "#,
+        exp: map![
+            "b" => "string",
+            "x" => "[{ a: string }]",
+            "y" => "[{ a: int, b: float }]",
+            "z" => "[{ a: int, b: float }]",
+        ],
+    }
+}
+
+#[test]
+fn labels_unbound() {
+    test_infer! {
+        config: AnalyzerConfig{
+            features: vec![Feature::LabelPolymorphism],
+            ..AnalyzerConfig::default()
+        },
+        env: map![
+            "f" => "(<-tables: [{ A with B: C }], ?value: D) => [{ A with B: D }]
+                where B: Label
+                "
+        ],
+        src: r#"
+            x = [{ a: 1, b: 2.0 }] |> f(value: "x")
+        "#,
+        exp: map![
+            "x" => "[{ a: string, b: float }]",
+        ],
+    }
+}
```

Comment:
This test case confuses me, what is the purpose of the unbounded `B` label? Since `B` is never defined what is its purpose?

Is the idea to show that it doesn't error? Should it error?

Reply:
Yeah, this should error. I think I wrote this test case early on and it happens to still work.